src/metatensor/models/soap_bpnn/model.py

```diff
@@ -92,6 +86,50 @@ def forward(self, features: TensorMap) -> TensorMap:
         return TensorMap(keys=features.keys, blocks=new_blocks)
 
 
+class LinearMap(torch.nn.Module):
+    def __init__(self, all_species: List[int], n_inputs: int) -> None:
+        super().__init__()
+
+        # Build a neural network for each species
+        layer_per_species = []
+        for _ in all_species:
+            layer_per_species.append(torch.nn.Linear(n_inputs, 1))
+
+        # Create a module dict to store the neural networks
+        self.layers = torch.nn.ModuleDict(
+            {
+                str(species): layer
+                for species, layer in zip(all_species, layer_per_species)
+            }
+        )
+
+    def forward(self, features: TensorMap) -> TensorMap:
+        # Create a list of the blocks that are present in the features:
+        present_blocks = [
+            int(features.keys.entry(i).values.item())
+            for i in range(features.keys.values.shape[0])
+        ]
+
+        new_blocks: List[TensorBlock] = []
+        for species_str, layer in self.layers.items():
+            species = int(species_str)
+            if species not in present_blocks:
+                pass  # continue is not accepted by torchscript here
+            else:
```

Comment:
Can't you turn this around 

```suggestion
            if species in present_blocks:
```